posts/2023-02-17-using-liberty-buildpack-on-azure-and-google-cloud.adoc

```diff
@@ -0,0 +1,108 @@
+
+```
+az acr pack build 
+    --registry mycr 
+    --pull --builder paketobuildpacks/builder:full 
+    --image myapp 
+    ~/myappsourcedir
+```
+
+== Using Google Cloud
+Use the link:https://cloud.google.com/docs/buildpacks/build-application#remote_builds[gcloud builds] command to build and push a container image using the built-in cloud native buildpack support.  
+Follow the link:https://cloud.google.com/docs/buildpacks/build-application#before-you-begin[Before you begin] instructions to setup your environment.  
+
+The basic command syntax is:
+```
+gcloud builds submit --pack image=LOCATION-docker.pkg.dev/PROJECT_ID/REPO_NAME/IMAGE_NAME
+```
+
+In the following example, `us-east1` is the LOCATION, `paketobuildpacks` is the PROJECT_ID, `paketobuildpacktest`` is the REPO_NAME and `javatest` is the IMAGE_NAME.  The command is executed from the application's root directory.  
+
+```
+gcloud builds submit --pack 
+    image=us-east1-docker.pkg.dev/paketobuildpacks/paketobuildpacktest/javatest,builder=paketobuildpacks/builder:full
+
+```
+
+== Other environment variables you can set in the `project.toml` file
+BP_LIBERTY_INSTALL_TYPE::
+Specifies the link:https://github.com/paketo-buildpacks/liberty#install-types[Install type] of Liberty. Open Liberty (value of `ol`) is the default.  Use `wlp` for WebSphere Liberty.  
+
+BP_LIBERTY_PROFILE::
+Specifies which liberty profile to install. Valid profiles for Liberty are documented link:https://github.com/paketo-buildpacks/liberty#profiles[in the buildpacks documentation].  The default profile is `kernel`.  
+
+BP_LIBERTY_FEATURES::
+Specifies a space-separated list of Liberty features to be installed with the Liberty runtime. Supports any valid Liberty feature.
+
+== Taking full advantage of all what the Paketo Buildpack for Liberty has to offer
+Some features of the Paketo Buildpack for Liberty are not easily available to Azure or Google Cloud. Features like link:https://github.com/paketo-buildpacks/liberty/blob/main/docs/installing-ifixes.md[installing iFixes], link:https://github.com/paketo-buildpacks/liberty#using-custom-features[custom features], and installing from a link:https://github.com/paketo-buildpacks/liberty#building-from-a-packaged-server[packaged server] or link:https://github.com/paketo-buildpacks/liberty#building-from-a-liberty-server[server directory] aren't available when you use the built-in support to create the container image.
+
+For these features, you can use the `pack build` CLI to create the container image, tag and push the image to an external container registry. Then, use Azure or Google Cloud to deploy and manage your container by pulling your container image from the container registry.
```

Comment:
Can we provide a link here to documentation about using pack build cli with Liberty? Maybe in the liberty paketo doc or https://openliberty.io/blog/2022/04/01/cloud-native-liberty-buildpack.html

Reply:
Yes, good suggestion.  The link to the blog is the best one....really the only place when we have any sort of documentation on using the pack cli with Liberty.   I'll add a link.

Reply:
Update made.